scripts/cluster-install/install.py

```diff
@@ -0,0 +1,392 @@
+    instances_dict['factory'] = [factory]
+    if len(instances_details) > 10:
+        load_balancer = _get_vm(instances_details, 'load_balancer', key_path,
+                                vm_username)
+        instances_dict['load_balancer'] = [load_balancer]
+    return instances_dict
+
+
+def _install_load_balancer(instances_dict, key_path):
+    logging.info('Installing load balancer')
+    load_balancer = instances_dict['load_balancer'][0]
+    managers_details = ''
+    scp_local_to_remote(key_path, load_balancer, LOCAL_INSTALL_CLUSTER_CERTS,
+                        REMOTE_PARENT_DIRECTORY)
+    scp_local_to_remote(key_path, load_balancer, 'install_load_balancer.sh',
+                        REMOTE_PARENT_DIRECTORY)
+    for manager in instances_dict['manager_nodes']:
+        managers_details += (manager.private_ip + ' ' +
+                             manager.public_ip + ' ')
+    install_command = 'sudo bash {script_path} {managers_details} {directory}' \
+        .format(script_path=os.path.join(REMOTE_PARENT_DIRECTORY,
+                                         'install_load_balancer.sh'),
+                managers_details=managers_details,
+                directory=REMOTE_PARENT_DIRECTORY)
+    stdin, stdout, stderr = load_balancer.exec_command(install_command)
+    print(stdout.read())
+    return
+
+
+def _create_install_cluster_directory(license_path, download_link):
+    logging.info('Creating `install_cluster` directory')
+    os.system('rm -rf {}_old'.format(LOCAL_INSTALL_CLUSTER))
+    if os.path.exists(LOCAL_INSTALL_CLUSTER):
+        os.system('mv {0} {0}_old'.format(LOCAL_INSTALL_CLUSTER))
+    os.mkdir(LOCAL_INSTALL_CLUSTER)
+    os.mkdir(LOCAL_INSTALL_CLUSTER + '/certs')
+    _download_manager_and_create_license(license_path, download_link)
+
+
+def _parse_command():
+    parser = argparse.ArgumentParser(description='Installing an Active-Active '
+                                                 'manager cluster')
+    parser.add_argument('--config-path', action='store', type=str,
+                        required=True, help='The config.yaml file path')
+    parser.add_argument('--do-not-clean-on-failure', dest='clean',
+                        action='store_false', default=True,
+                        help='Pass this flag if you do not want to clean '
+                             'your Openstack environment on failure')
+    return parser.parse_args()
+
+
+def _delete_factory_vm_from_openstack_env(connection, environment_ids_dict):
+    delete_vm(connection, environment_ids_dict['factory'])
+    del environment_ids_dict['factory']
+
+
+def _create_environment_ids_file(environment_ids_dict):
+    with open('environment_ids.yaml', 'w') as f:
+        yaml.dump(environment_ids_dict, f)
+
+
+def _show_successful_installation_message(start_time, end_time):
+    logging.info('Successfully installed an Active-Active cluster in %.2f'
+                 ' minutes' % round(((end_time - start_time) / 60), 2))
```

Comment:
the `round` is somewhat useless if we already use the `%.2f` format

Reply:
additionally, (noting since this is already using %-formatting instead of str.format) note that the preferred way should be
```python
logger.info('abc %.2f', 1.23)
```
instead of 
```python
logger.info('abc %.2f' % 1.23)
```
(ie. providing the format arguments as separate arguments to logging functions, instead of formatting them yourself)
this way, the logging system will only do the string formatting (which is possibly slow to run) if it decides that the message will actually be emitted (think `logger.debug` when the logging level is lower than debug)

doesn't matter much for this specific case, since this message will always be shown, but as a general hint

Reply:
Done, thanks a lot.